src/netaddress.cpp

```diff
@@ -822,68 +854,27 @@ bool CSubNet::Match(const CNetAddr &addr) const
 {
     if (!valid || !addr.IsValid() || network.m_net != addr.m_net)
         return false;
-    for(int x=0; x<16; ++x)
-        if ((addr.ip[x] & netmask[x]) != network.ip[x])
+    assert(network.m_addr.size() == addr.m_addr.size());
+    for (size_t x = 0; x < addr.m_addr.size(); ++x)
+        if ((addr.m_addr[x] & netmask[x]) != network.m_addr[x])
```

Comment:
Braces, please.

(for most rules I don't care that strongly whether you stick to the existing/surrounding style vs. follow the new style, but a braceless `if` has actually been part of the cause of a nontrivial security bug, see https://www.imperialviolet.org/2014/02/22/applebug.html)